File: output/csharp/QueryDsl/FullText/MatchBoolPrefix/MatchBoolPrefixQuery.cs
using Nest.CommonOptions;
using Nest.QueryDsl;
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.QueryDsl {

	public class MatchBoolPrefixQuery  {
		
		[DataMember(Name="analyzer")]
		public string Analyzer { get; set; }


		[DataMember(Name="fuzziness")]
		public Fuzziness Fuzziness { get; set; }


		[DataMember(Name="fuzzy_rewrite")]
		public MultiTermQueryRewrite FuzzyRewrite { get; set; }


		[DataMember(Name="fuzzy_transpositions")]
		public bool FuzzyTranspositions { get; set; }


		[DataMember(Name="max_expansions")]
		public int MaxExpansions { get; set; }


		[DataMember(Name="minimum_should_match")]
		public MinimumShouldMatch MinimumShouldMatch { get; set; }


		[DataMember(Name="operator")]
		public Operator Operator { get; set; }


		[DataMember(Name="prefix_length")]
		public int PrefixLength { get; set; }


		[DataMember(Name="query")]
		public string Query { get; set; }

	}
}

File: specification/specs/query_dsl/full_text/match_bool_prefix/MatchBoolPrefixQuery.ts
@class_serializer("FieldNameQueryFormatter`2")
class MatchBoolPrefixQuery {
	analyzer: string;
	fuzziness: Fuzziness;
	fuzzy_rewrite: MultiTermQueryRewrite;
	fuzzy_transpositions: boolean;
	max_expansions: integer;
	minimum_should_match: MinimumShouldMatch;
	operator: Operator;
	prefix_length: integer;
	query: string;
  boost: float;
}
